answerkey/state/11.answer.scala

```diff
@@ -14,7 +14,10 @@ object Candy {
         Machine(false, candy, coin + 1)
       case (Turn, Machine(false, candy, coin)) =>
         Machine(true, candy - 1, coin)
-    })))
+    }
+
+  def simulateMachine(inputs: List[Input]): State[Machine, (Int, Int)] = for {
+    _ <- sequence(inputs map update map modify)
```

Comment:
There is a function called `traverse` in the book that combines `map` and `sequence` into one pass, and you could also combine all those `map` calls into one since `inputs map update map modify` is equivalent to `inputs map (modify compose update)`.